### PR TITLE
Add preference to delay start of video

### PR DESCRIPTION
**Pull request by JagerSprinkles**
**Changes**

This updates the delay when starting the video player to be user configurable. Having this as configurable will help people who have displays that are slow to change resolution or refresh rate. I have two displays and a soundbar that take longer than the 0.75 seconds default to change format or mode. With this setting the first few seconds of the video and audio are not missed.

**Issues**

None I can find.

### CODE REVIEW

File: app/src/main/java/org/jellyfin/androidtv/preference/UserPreferences.kt [hunk 1]
```diff
@@ -200,6 +200,11 @@ class UserPreferences(context: Context) : SharedPreferenceStore(
 		 * Whether items shown in the screensaver are required to have an age rating set.
 		 */
 		var screensaverAgeRatingRequired = booleanPreference("screensaver_agerating_required", true)
+
+		/**
+		 * Delay when starting video playback after loading the video player.
+		 */
+		var videoStartDelay = longPreference("video_start_delay", 750)
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/preference/UserPreferences.kt:207]: Could default to just `0`. There really is no reason to delay playback by default.

File: app/src/main/java/org/jellyfin/androidtv/ui/playback/PlaybackController.java [hunk 2]
```diff
@@ -646,7 +646,7 @@ public void run() {
                 if (mVideoManager != null)
                     mVideoManager.start();
             }
-        }, 750);
+        },  userPreferences.getValue().get(UserPreferences.Companion.getVideoStartDelay()));
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/ui/playback/PlaybackController.java:649]: The comment above (about why this is delayed) is inaccurate and could be removed. Additionally a fast-path when there is no delay would be nice to have.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -516,6 +516,7 @@
     <string name="past_24_hours">Past 24 hours</string>
     <string name="prefer_exoplayer_ffmpeg">Prefer FFmpeg for audio playback</string>
     <string name="prefer_exoplayer_ffmpeg_content">Use FFmpeg to decode audio, even if platform codecs are available.</string>
+    <string name="lbl_video_start_delay">Video Start Delay</string>
```
Comment (nielsvanvelzen) [inline app/src/main/res/values/strings.xml:519]:
```suggestion
    <string name="lbl_video_start_delay">Video start delay</string>
```

Comment (nielsvanvelzen) [reply app/src/main/res/values/strings.xml:519]: Also the `lbl_` prefix isn't needed

**Review by nielsvanvelzen — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/jellyfin/androidtv/preference/UserPreferences.kt (hunk 1 above)
Comment (JagerSprinkles) [reply app/src/main/java/org/jellyfin/androidtv/preference/UserPreferences.kt]: This has been updated to 0 now.

File: app/src/main/java/org/jellyfin/androidtv/ui/playback/PlaybackController.java (hunk 2 above)
Comment (JagerSprinkles) [reply app/src/main/java/org/jellyfin/androidtv/ui/playback/PlaybackController.java]: By fast-path I assumed you meant to add a condition to skip the mHandler.postDelayed logic if the delay is set to 0. I have added a check for this.

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (JagerSprinkles) [reply app/src/main/res/values/strings.xml]: I have updated the string name and value as requested.

File: app/src/main/java/org/jellyfin/androidtv/ui/playback/PlaybackController.java [hunk 4]
```diff
@@ -639,14 +639,20 @@ private void startItem(BaseItemDto item, long position, StreamInfo response) {
...
 
-        //wait a beat before attempting to start so the player surface is fully initialized and video is ready
-        mHandler.postDelayed(new Runnable() {
-            @Override
-            public void run() {
-                if (mVideoManager != null)
-                    mVideoManager.start();
+        //set video start delay
+        long videoStartDelay = userPreferences.getValue().get(UserPreferences.Companion.getVideoStartDelay());
+        if (mVideoManager != null) {
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/ui/playback/PlaybackController.java:644]: This null cehck should only happen in the delayed function as we need to verify the manager is not disposed. We already know it is not null when calling it directly.

**Review by nielsvanvelzen — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/jellyfin/androidtv/ui/playback/PlaybackController.java (hunk 2 above)
Comment (nielsvanvelzen) [reply app/src/main/java/org/jellyfin/androidtv/ui/playback/PlaybackController.java]: Yup that's what I meant!

File: app/src/main/java/org/jellyfin/androidtv/ui/playback/PlaybackController.java (hunk 4 above)
Comment (JagerSprinkles) [reply app/src/main/java/org/jellyfin/androidtv/ui/playback/PlaybackController.java]: I have updated the null check for only when a delay has been set.

**Review by nielsvanvelzen — APPROVED** [review]
Awesome, thank you! I've pushed some final changes (see last commit).

**Comment by JagerSprinkles** [comment]
Thanks for the little correction in that last commit!

**Pull request merged**